sklearn/metrics/_pairwise_distances_reduction/_base.pyx.tp

```diff
@@ -416,8 +445,17 @@ cdef class BaseDistanceReducer{{name_suffix}}:
     ) nogil:
         """Initialize datastructures just before the _compute_and_reduce_distances_on_chunks.
 
-        This is eventually used to upcast Y[Y_start:Y_end] to 64bit.
-        """
+        In this method, EuclideanDistance specialisations of subclass of
+        BaseDistancesReduction _must_ call:
+
+        self.gemm_term_computer._parallel_on_Y_pre_compute_and_reduce_distances_on_chunks(
+            X_start, X_end, Y_start, Y_end, thread_num,
+        )
+
+        to ensure the proper upcast of Y[Y_start:Y_end] to float64 prior
+        to the reduction with float64 accumulator buffers when Y.dtype is
+        float32.
+        """""
```

Comment:
Sorry, probably my fault:

```suggestion
        """"
```

Reply:
I think this originates from a faulty copy-paste of mine.